lib/internal/process/next_tick.js

```diff
@@ -88,20 +109,58 @@ function setupNextTick() {
     }
   }
 
+  // TODO(trevnorris): Using std::stack is much slower here than was the
+  // Float64Array stack used in a previous implementation. Problem is the
+  // Float64Array stack was a bit brittle. Investigate how to harden that
+  // implementation and possibly reintroduce it.
```

Comment:
The performance hit comes from entering C++. The previous implementation used a `Float64Array` to store the stack and would rotate it if the stack filled. It was quite a pain, but incurred very little overhead.